sdk/appconfiguration/app-configuration/src/secretReference.ts

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { errorMessageForUnexpectedSetting } from "./internal/helpers";
+import { JsonSecretReferenceValue } from "./internal/jsonModels";
+import { ConfigurationSetting, ConfigurationSettingParam } from "./models";
+
+/**
+ * content-type for the secret reference.
+ */
+export const secretReferenceContentType =
+  "application/vnd.microsoft.appconfig.keyvaultref+json;charset=utf-8";
+
+/**
+ * Necessary fields for updating or creating a new secret reference.
+ */
+export interface SecretReferenceValue {
+  /**
+   * Id for the secret reference.
+   */
+  secretId: string;
+}
+
+/**
+ * @internal
+ */
+export const SecretReferenceHelper = {
```

Comment:
why are we exporting this as an object instead of exporting the two methods directly?

Reply:
This is not exported to the public, it is internal only.
We don't export it in the index.ts.
![image](https://user-images.githubusercontent.com/10452642/118729032-050e8500-b7ea-11eb-9d4b-466c831245d4.png)

Reply:
Oh I'm not worried about the public shape, it's more that it doesn't seem to make sense to me internally to export it this way from this module file.

I would have expected something like this:

```ts
export function fromConfigurationSetting();
export function toConfigurationSettingParam();
```

Seems to look much nicer than exporting an object called `SecretReferenceHelper` with those two functions as properties. 

Even more so since you end up exporting `fromConfigurationSetting` twice: once as `SecretReferenceHelper.fromConfigurationSetting` and once as `parseSecretReference `

Reply:
Okay, removed `fromConfigurationSetting` from the Helper (both feature flag and secret reference).
And kept `parseSecretReference` and `parseFeatureFlag` only!